safe_control_gym/envs/gym_pybullet_drones/quadrotor.py

```diff
@@ -798,7 +798,13 @@ def _get_observation(self):
             obs = self.disturbances['observation'].apply(obs, self)
 
         # Concatenate goal info (references state(s)) for RL.
-        obs = self.extend_obs(obs, self.ctrl_step_counter+1)
+        # Plus two because ctrl_step_counter has not incremented yet, and we want to return the obs (which would be
+        # ctrl_step_counter + 1 as the action has already been applited), and the next state (+ 2) for the RL to see
+        # the next state.
+        if self.at_reset:
+            obs = self.extend_obs(obs, self.ctrl_step_counter+1)
```

Comment:
Again, I think the second argument would always evaluate to just 1

Reply:
Fixed.